### ISSUE TITLE
Rename "Move to Trash/Recycle Bin" back to "Delete" on Windows

### ISSUE THREAD

**Issue by Xrayez** [godotengine/godot-proposals#2463]
### Describe the project you are working on

[Goost](https://github.com/goostengine/goost) Godot Engine extension.

### Describe the problem or limitation you are having in your project

See previous decision to rename "Delete" to "Move to Trash" in godotengine/godot#41263 for all platforms, and then the decision to rename "Move to Trash" to "Move to Recycle Bin" on Windows in godotengine/godot#47004, and corresponding proposal: #2460.

I think that "Move to Recycle Bin" breaks user expectations for most Windows users (including myself). I realize that it does what it says on the tin, but it's not what a typical Windows user would look for, which breaks the workflow for existing Windows users (especially for new Godot users on Windows).

If we aim for explicitness, then we've lost clarity/intuitiveness, because as a Windows user, it's not immediately clear what menu option is responsible for this action, because Windows users expect it to be "Delete", and not "Move to Recycle Bin". It's also quite verbose when comparing to Linux/macOS variants.

Current `master` version (4.0-dev):

![image](https://user-images.githubusercontent.com/17108460/111091891-0fd44180-853d-11eb-935d-f7a99abdce2e.png)

I also don't like the fact that "Move to Recycle Bin" is too close to "Move to..." action, making it even more confusing sometimes.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

On Windows, the menu option responsible for deleting files should be renamed back to "Delete" as it was before.

To compare, I've only recently installed a Linux distro, and I got somewhat puzzled until I figured out that to "delete" a file/folder, I have to look for "Move to Trash" instead.

#### Windows

![image](https://user-images.githubusercontent.com/17108460/111079335-3e80f680-8502-11eb-9a33-3ebfb3edbb5e.png)

#### Linux

![image](https://user-images.githubusercontent.com/17108460/111089384-52dde700-8534-11eb-90a2-79229395e79e.png)

#### macOS

![macOS](https://user-images.githubusercontent.com/7645683/111079701-cb787f80-8503-11eb-9281-9a8641f5a1b8.png)

With godotengine/godot#47004 merged, we have corresponding names for both Linux macOS, but we have a mismatch on Windows, more severe than it was before ("Move to Trash").

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
```diff
diff --git a/editor/filesystem_dock.cpp b/editor/filesystem_dock.cpp
index 43bfccc656..129a26ff92 100644
--- a/editor/filesystem_dock.cpp
+++ b/editor/filesystem_dock.cpp
@@ -2720,10 +2720,12 @@ FileSystemDock::FileSystemDock(EditorNode *p_editor) {
        ED_SHORTCUT("filesystem_dock/duplicate", TTR("Duplicate..."), KEY_MASK_CMD | KEY_D);

 #if defined(WINDOWS_ENABLED)
-       // TRANSLATORS: This string is only used on Windows, as it refers to the system trash
-       // as "Recycle Bin" instead of "Trash". Make sure to use the translation of "Recycle Bin"
-       // recommended by Microsoft for the target language.
-       ED_SHORTCUT("filesystem_dock/delete", TTR("Move to Recycle Bin"), KEY_DELETE);
+       // Note that the "Delete" action is actually responsible for moving files
+       // to "Recycle Bin" by default instead of deleting files altogether, so we
+       // use "Delete" term here rather than more explicit "Move to Recycle Bin",
+       // otherwise this will break user expectations. Once Microsoft decides to
+       // rename it to "Move to Recycle Bin", then it's safe to rename it in Godot.
+       ED_SHORTCUT("filesystem_dock/delete", TTR("Delete"), KEY_DELETE);
 #elif defined(OSX_ENABLED)
        // TRANSLATORS: This string is only used on macOS, as it refers to the system trash
        // as "Bin" instead of "Trash". Make sure to use the translation of "Bin"
```

### If this enhancement will not be used often, can it be worked around with a few lines of script?
[Have to patch the engine](https://goost.readthedocs.io/en/latest/usage/configuring.html#patching) to rename it.

I'm sorry if I'm just being vocal about this and it may be obvious to you, but I'd like to understand what are we aiming for. Because current decision does not make sense to me: we shouldn't break established user expectations just for Windows users.

To resolve this for all platforms, perhaps Godot should just pick another word for this action, like "Remove", and not to depend on platform implementation.

### Is there a reason why this should be core and not an add-on in the asset library?
Godot should follow OS names for similar actions even if it does not make sense to Linux/macOS users, or do not follow them at all. Microsoft hadn't decided to rename the menu option to "Move to Recycle Bin", so I don't see a reason why Godot should break user expectations just for the sake of consistency with Linux/macOS.

Lets have a discussion. Please don't just downvote this proposal if you disagree (if you do, please explain why). I treat "small things" on the same level as "big things". Thank you!

I'd also create a discussion instead of creating yet another proposal, but GIP has no GitHub Discussions enabled here: #2069. If we do enable this feature eventually, feel free to transfer this "proposal" to "discussion".

**Comment by LightningAA** [comment]
I like "Move to Recycle Bin" more than "Delete". "Delete" because I think it's more clear what it does, but I think it really doesn't matter.

**Comment by Zireael07** [comment]
You said that Move to Recycle Bin breaks expectations on Windows. The other person, who requested the change to "recycle bin", clearly thought "delete" breaks expectations. I think for each option we'll see people in favor of that - and unfortunately there seems to be no neutral wording...

**Comment by me2beats** [comment]
I think `Delete` is ok because
- It is less noisy
- Info that the file is being moved to the trash exists in the confirmation dialog anyway
![trash](https://user-images.githubusercontent.com/16458555/111125044-fe1d8900-8592-11eb-8544-27251bad8bc5.png)
- The presence of a `trash icon` assumes that the file is being removed to the trash
![trash icon](https://user-images.githubusercontent.com/16458555/111126049-14781480-8594-11eb-9c12-5fc31fc93595.png)

- I know how sad it can be to delete files without being able to recover them.
But I don't know what problems might arise if the user decides that a file is permanently deleted but it doesn't.
in theory, it could have something to do with data security if someone else has access to the trash. But I think this is an extremely rare case.

**Comment by Xrayez** [comment]
> You said that Move to Recycle Bin breaks expectations on Windows. The other person, who requested the change to "recycle bin", clearly thought "delete" breaks expectations. I think for each option we'll see people in favor of that - and unfortunately there seems to be no neutral wording...

If you're talking about #2460, I think that the person wasn't happy with "Move to Trash" name, and at least suggested to rename it to "Move to Recycle bin". There was no comment on whether the "Delete" name is better/obvious or not. That user would likely be happy and feel alright if the option remained to be "Delete" on Windows, but since there's a realization that Godot developers won't likely change it to "Delete", the author probably decided to go with the "least resistance" route. The user is also likely new to Godot so there was no awareness that Godot previously had "Delete" across all operating systems in previous versions (throughout existence of Godot).

I really think that godotengine/godot#41263 just missed the fact that not all operating systems use "Move to Trash" name for this action, or Godot simply chooses to be more on the side of Linux-like operating systems, given that Godot itself is open-source.

People who complain that "Delete" is not explicit enough are the ones who use either Linux or macOS for Godot development on a daily basis. But we don't really know what Windows users think about this (likely most devs use Linux to develop Godot as well), and perhaps we may find this out with this proposal.

In any case, I don't see anyone complaining about this in the past and there was nothing to solve, it was simply a change for the sake of explicitness, like what you can see in #639, for instance. And I think that godotengine/godot#47004 went a little bit too far when we talk about Windows, because now we have a mix of both Linux-style and Windows-style naming conventions, rather than sticking to what the operating system chooses for this action by default, simply as that.

> I like "Move to Recycle Bin" more than "Delete". "Delete" because I think it's more clear what it does, but I think it really doesn't matter.

Again, this is more about consistency. If Microsoft chooses to rename the equivalent action to "Move to Recycle Bin", then the rename would be fine. I wouldn't even complain because I'd be already accustomed to recognize it immediately. This is currently the case for Linux and macOS users, but *not* Windows...

I really think that this does not matter much either, but I simply felt that my opinion was simply ignored when I attempted to discuss this directly in godotengine/godot-proposals#2460. Instead, *some* discussion happened directly in the PR, instead of discussing this in the proposal, and then I got surprised that it got merged like that. 😮

I'd probably wouldn't create this and #2462 proposals if I'd receive at least some feedback from the core developers.

**Comment by pycbouh** [comment]
I personally don't like `Move to` instead of `Delete` simply because it takes more time to locate the option, especially because it now starts with the same text as the actual `Move to` action. That said, "moving to recycle bin" exists in Windows glossary, so it's not a completely foreign phrase.

> I really think that this does not matter much either, but I simply felt that my opinion was simply ignored when I attempted to discuss this directly in #2460. Instead, some discussion happened directly in the PR, instead of discussing this in the proposal, and then I got surprised that it got merged like that. 😮
> 
> I'd probably wouldn't create this and #2462 proposals if I'd receive at least some feedback from the core developers (in a timely manner, before the PR got merged, and without me pinging the author to figure out the rationale as to why).

Your grief is with https://github.com/godotengine/godot/pull/41263, not with #2460. And that PR was in limbo for 3 months. Then the decision was made and #2460 is just a small enhancement on top of that.

**Comment by Xrayez** [comment]
> Your grief is with [godotengine/godot#41263](https://github.com/godotengine/godot/pull/41263), not with #2460. And that PR was in limbo for 3 months. Then the decision was made and #2460 is just a small enhancement on top of that.

Yes and no, it made it worse in both cases the way I see it. I realize there was a change for this in the past and I accepted it (because it was applied to all platforms, more or less agnostic to each platform) so I didn't express my disagreement 3 months ago. But now, this is a totally different story, that change lead to a problem which did not exist before.

**Comment by rick551a** [comment]
Yes, bring back 'Delete'. 
It's standard, corresponds to 'del' key on the keyboard.
Maybe a (default unchecked) checkbox could be placed on the 'Confirm delete' dialog which asks 'Permanently delete?'.
If unchecked, send the files to Trash/Recycle Bin, otherwise permanently delete.

**Comment by smix8** [comment]
Windows uses "delete" or similar, short terms in all localized menu actions to remove files (and by default moves them in the system recycle bin). Not only is "Move to Recycle Bin" in Godot not true for Windows user expectations (my system "Recycle Bin" is still empty and Godot has deleted the file), it is also a convoluted way saying "delete" from the perspective of a Windows peasant.

**Comment by pycbouh** [comment]
> Not only is "Move to Recycle Bin" in Godot not true for Windows user expectations (my system "Recycle Bin" is still empty and Godot has deleted the file)

That's some kind of issue on your end. The motivation to rename was precisely because Godot does not permanently delete files, but moves them to the bin. If this menu option permanently deletes a file for you, please report it.

**Comment by mrjustaguy** [comment]
Move to Trash is MUCH clearer on what it's doing. see https://github.com/godotengine/godot-proposals/issues/2460#issuecomment-800822287

TLDR - One can have windows just skip the Recycle bin part and Delete it (non default windows setting) or have windows just move to Recycle bin (default windows setting) and both are "Delete" so for some it means actually killing the file even on windows, which is NOT the case as Godot just dumps into Recycle bin.

**Comment by Xrayez** [comment]
**smix8**, **pycbouh**

> > Not only is "Move to Recycle Bin" in Godot not true for Windows user expectations (my system "Recycle Bin" is still empty and Godot has deleted the file)
> 
> That's some kind of issue on your end. The motivation to rename was precisely because Godot does not permanently delete files, but moves them to the bin. If this menu option permanently deletes a file for you, please report it.

**mrjustaguy**

> Move to Trash is MUCH clearer on what it's doing. see [#2460 (comment)](https://github.com/godotengine/godot-proposals/issues/2460#issuecomment-800822287)

Well I've just tested https://github.com/godotengine/godot-proposals/issues/2460#issuecomment-800822287 and I think that's unexpected, so I created an issue for this in Godot now: godotengine/godot#47106.

**Comment by mrjustaguy** [comment]
Idk, It clearly states in the picture You provided in the bug report that It's moved to the Recycle bin so I don't see what's so unexpected about it being moved to trash, It actually killing it with windows when you're without a Recycle bin, that's a bug. possibly something that is unavoidable, which is like the only reason to Change it to Delete..

**Comment by mhilbrunner** [comment]
Windows uses 'Delete' everywhere, regardless of that moving things to the Recycle Bin.
I think Godot should follow that. 'Move to Recycle Bin' is also easy to misread as 'Move to...' at a glance, as stated.
